search/search-client-opensearch/src/main/java/io/camunda/search/os/transformers/query/TermQueryTransformer.java

```diff
@@ -24,6 +24,10 @@ public TermQuery apply(final SearchTermQuery value) {
     final var transformer = getFieldValueTransformer();
     final var fieldValue = value.value();
     final var tranformedFieldValue = transformer.apply(fieldValue);
-    return QueryBuilders.term().field(field).value(tranformedFieldValue).build();
+    return QueryBuilders.term()
+        .field(field)
+        .value(tranformedFieldValue)
+        .caseInsensitive(value.caseInsensitive())
```

Comment:
I wonder why is we can use primitive boolean for `caseInsensitive`, to avoid handling the null case

Reply:
Null is fine here. This is handled by OS and ES client correctly.